### PR TITLE
graphics utility for more than one ROI

### PR DESCRIPTION
**Pull request by davlars**
Added utility for visualization of multiple or nested ROIs. (Previously `show_both` has now been extended with `show_all`)

### CODE REVIEW

File: odl_multigrid/graphics.py [hunk 1]
```diff
@@ -102,6 +102,48 @@ def show(data, eps=0.0):
     # TODO: set aspect from physical sizes
     ax.set_aspect('auto', 'box')
 
+def show_all(data, low = None, high = None):
```
Comment (kohr-h) [inline odl_multigrid/graphics.py]:
```suggestion
def show_all(data, low=None, high=None):
```

File: odl_multigrid/graphics.py [hunk 2]
```diff
@@ -102,6 +102,48 @@ def show(data, eps=0.0):
...
+    fig, ax = plt.subplots()
+    
+    ax.set_xlim(data[0].space.min_pt[0], data[0].space.max_pt[0])
+    ax.set_ylim(data[0].space.min_pt[1], data[0].space.max_pt[1])
+
+    if low is None:
+        low = min([np.min(data[0]), np.min(data[1])])
+    if high is None:
+        high = max([np.max(data[0]), np.max(data[1])])
+
```
Comment (kohr-h) [inline odl_multigrid/graphics.py]: Remove extra line

File: odl_multigrid/graphics.py [hunk 3]
```diff
@@ -102,6 +102,48 @@ def show(data, eps=0.0):
...
+def show_all(data, low = None, high = None):
+    """Show all reconstructed resolutions in a single image. For multiple ROIs"""
+    # Lazy import, can be slow
+    import matplotlib as mpl
+    import matplotlib.pyplot as plt
+    from matplotlib.image import BboxImage
+    from matplotlib.transforms import Bbox, TransformedBbox
+
+    fig, ax = plt.subplots()
+
```
Comment (kohr-h) [inline odl_multigrid/graphics.py:122]: Remove extra line

File: odl_multigrid/graphics.py [hunk 4]
```diff
@@ -102,6 +102,48 @@ def show(data, eps=0.0):
     # TODO: set aspect from physical sizes
     ax.set_aspect('auto', 'box')
 
+def show_all(data, low = None, high = None):
+    """Show all reconstructed resolutions in a single image. For multiple ROIs"""
```
Comment (kohr-h) [inline odl_multigrid/graphics.py]: Maybe a word on what ``data`` is expected to be?

**Review by kohr-h — COMMENTED** [review]
Thanks! Small stuff only.

File: odl_multigrid/graphics.py (hunk 2 above)
Comment (davlars) [reply odl_multigrid/graphics.py]: Modified in a2a42d3

File: odl_multigrid/graphics.py (hunk 3 above)
Comment (davlars) [reply odl_multigrid/graphics.py:122]: Modified in a2a42d3

File: odl_multigrid/graphics.py (hunk 4 above)
Comment (davlars) [reply odl_multigrid/graphics.py]: Yes. Added info in a2a42d3 in the form of

- `data: list of reconstructions, with background given as data[0] and
              subsequent ROI reconstructions given as data[1], data[2], ...`
- `low: lower bound of dynamic range. If 'None' set to min(data)`
- `high: upper bound of dynamic range. If 'None' set to max(data)`

File: odl_multigrid/graphics.py (hunk 3 above)
Comment (davlars) [reply odl_multigrid/graphics.py:122]: ...I think? :confused: Or did I miss which line you were after?

**Comment by kohr-h** [comment]
Thanks!

**Pull request merged**